sql/src/main/java/org/apache/apex/malhar/sql/codegen/BeanClassGenerator.java

```diff
@@ -0,0 +1,796 @@
+          hashCodeNode.instructions.add(new InsnNode(Opcodes.LUSHR));
+          hashCodeNode.instructions.add(new InsnNode(Opcodes.LXOR));
+          hashCodeNode.instructions.add(new InsnNode(Opcodes.L2I));
+          break;
+        case "double":
+          hashCodeNode.instructions
+            .add(new MethodInsnNode(Opcodes.INVOKESTATIC, "java/lang/Double", "doubleToLongBits", "(D)J", false));
+          hashCodeNode.instructions.add(new InsnNode(Opcodes.DUP2));
+          hashCodeNode.instructions.add(new VarInsnNode(Opcodes.LSTORE, 2));
+          hashCodeNode.instructions.add(new VarInsnNode(Opcodes.LLOAD, 2));
+          hashCodeNode.instructions.add(new IntInsnNode(Opcodes.BIPUSH, 32));
+          hashCodeNode.instructions.add(new InsnNode(Opcodes.LUSHR));
+          hashCodeNode.instructions.add(new InsnNode(Opcodes.LXOR));
+          hashCodeNode.instructions.add(new InsnNode(Opcodes.L2I));
+          break;
+        default:
+          String objectOwnerType = fieldType.replace('.', '/');
+          LabelNode nullNode = new LabelNode();
+          LabelNode continueNode = new LabelNode();
+          hashCodeNode.instructions.add(new JumpInsnNode(Opcodes.IFNULL, nullNode));
+          hashCodeNode.instructions.add(new VarInsnNode(Opcodes.ALOAD, 0));
+          hashCodeNode.instructions.add(new FieldInsnNode(Opcodes.GETFIELD, classNode.name, fieldName, fieldJavaType));
+          hashCodeNode.instructions
+            .add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, objectOwnerType, "hashCode", "()I", false));
+          hashCodeNode.instructions.add(new JumpInsnNode(Opcodes.GOTO, continueNode));
+          hashCodeNode.instructions.add(nullNode);
+          hashCodeNode.instructions.add(new InsnNode(Opcodes.ICONST_0));
+          hashCodeNode.instructions.add(continueNode);
+          break;
+      }
+      hashCodeNode.instructions.add(new InsnNode(Opcodes.IADD));
+      hashCodeNode.instructions.add(new VarInsnNode(Opcodes.ISTORE, 1));
+    }
+    hashCodeNode.instructions.add(new VarInsnNode(Opcodes.ILOAD, 1));
+    hashCodeNode.instructions.add(new InsnNode(Opcodes.IRETURN));
+
+    classNode.methods.add(hashCodeNode);
+  }
+
+  /**
+   * Adds a equals method to underlying class. <br>
+   * Algorithm is as follows: <br>
+   * <i><p>
+   * if (this == other) return true;
+   * if (other == null) return false;
+   * if (other is not instanceof <this class>) return false;
+   * for (field: all fields) {
+   * if (other.getField() != this.field) return false;
+   * }
+   * return true;
+   * </p></i>
+   * <br>
+   *
+   * @param classNode
+   * @param fieldList
+   * @throws JSONException
+   */
+  @SuppressWarnings("unchecked")
+  private static void addEqualsMethod(ClassNode classNode, List<TupleSchemaRegistry.SQLFieldInfo> fieldList)
+    throws JSONException
+  {
+    MethodNode equalsNode = new MethodNode(Opcodes.ACC_PUBLIC, "equals", "(Ljava/lang/Object;)Z", null, null);
```

Comment:
I checked Objects.equals... It has only this: 
    public static boolean equals(Object a, Object b) {
        return (a == b) || (a != null && a.equals(b));
    }

Its does not check for individual elements. Please correct if my understanding is incorrect.